### PR TITLE
feat: enforce async functions to have await

### PR DESCRIPTION
**Pull request by james-a-morris**
This PR enforces that all functions marked with `async` must have at least one `await` within them. This is for two reasons: firstly, it provides a layer of code consistency. Secondly (and more importantly), it can help prevent accidental cases where we should use await but don't.

### CODE REVIEW

**Comment by james-a-morris** [comment]
Waiting for https://github.com/across-protocol/sdk-v2/pull/383 to resolve this failed test.

File: src/dataworker/Dataworker.ts [hunk 1]
```diff
@@ -1873,7 +1873,7 @@ export class Dataworker {
     });
   }
 
-  async _updateExchangeRates(l1Tokens: string[], submitExecution: boolean): Promise<void> {
+  _updateExchangeRates(l1Tokens: string[], submitExecution: boolean): void {
```
Comment (nicholaspai) [inline src/dataworker/Dataworker.ts:1876]: Are there any calls to `_updateExchangeRates` that needlessly `await` that we can change?

File: src/relayer/Relayer.ts [hunk 2]
```diff
@@ -565,7 +565,7 @@ export class Relayer {
     return profitable ? preferredChainId : undefined;
   }
 
-  protected async computeRealizedLpFeePct(version: number, deposit: DepositWithBlock): Promise<BigNumber> {
+  protected computeRealizedLpFeePct(version: number, deposit: DepositWithBlock): BigNumber {
```
Comment (nicholaspai) [inline src/relayer/Relayer.ts:568]: Same question as https://github.com/across-protocol/relayer-v2/pull/930/files#r1327228115 but just want to double check that you've removed any `await` calls to this function

**Review by nicholaspai — APPROVED** [review]
A couple questions to spur second checks

File: scripts/utils.ts [hunk 3]
```diff
@@ -115,7 +115,7 @@ export function resolveHubChainId(spokeChainId: number): number {
  * @param contractName Name of the deployed contract.
  * @returns ethers Contract instance.
  */
-export async function getContract(chainId: number, contractName: string): Promise<Contract> {
+export function getContract(chainId: number, contractName: string): Contract {
```
Comment (mrice32) [inline scripts/utils.ts:118]: Looks like there's a call below in this file that awaits this function. Is there a linter rule that can prevent us from calling await on a non-promise?

File: src/dataworker/Dataworker.ts (hunk 1 above)
Comment (mrice32) [reply src/dataworker/Dataworker.ts:1877]: Would be great if this could be linted automatically.

**Review by mrice32 — APPROVED** [review]
LGTM! Would be great if the linter could catch cases where we needlessly await.

**Comment by james-a-morris** [comment]
This is no longer needed in favor of co-PRs that have been added to the SDK & Relayer

**Pull request closed without merging**